specification/dfp/resource-manager/Microsoft.DFP/preview/2021-02-01-preview/dfp.json

```diff
@@ -0,0 +1,782 @@
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Dfp/instances": {
+      "get": {
+        "description": "Lists all the Dedicated instances for the given subscription.",
+        "x-ms-examples": {
+          "Get details of a instance": {
+            "$ref": "./examples/listInstancesInSubscription.json"
+          }
+        },
+        "tags": [
+          "Instances"
+        ],
+        "operationId": "Instances_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DFPInstances"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Dfp/instances/{location}/checkNameAvailability": {
+      "post": {
+        "x-ms-examples": {
+          "Check name availability of an instance": {
+            "$ref": "./examples/checkNameAvailability.json"
+          }
+        },
+        "tags": [
+          "Instances"
+        ],
+        "operationId": "Instances_CheckNameAvailability",
+        "description": "Check the name availability in the target location.",
+        "parameters": [
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The region name which the operation will lookup into."
+          },
+          {
+            "name": "instanceParameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CheckInstanceNameAvailabilityParameters"
+            },
+            "description": "The name of the instance."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/CheckInstanceNameAvailabilityResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "OperationListResult": {
```

Comment:
>OperationListResult [](start = 5, length = 19)

please re-use common-types/resource-management/v2/types.json#OperationListResult instead of redefining your own .. and for other common types like Operation and Resource below